app/src/main/java/com/infomaniak/swisstransfer/ui/utils/ContextExt.kt

```diff
@@ -105,6 +108,20 @@ fun Context.shareText(text: String) {
     safeStartActivity(Intent.createChooser(intent, null))
 }
 
+fun Context.openFile(uri: Uri) {
+    val intent = Intent(Intent.ACTION_VIEW).also {
+        it.data = uri
+        it.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        it.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+    }
+    try {
+        startActivity(intent)
+    } catch (e: ActivityNotFoundException) {
+        @OptIn(UnreliableToastApi::class)
+        toast(msgResId = R.string.startActivityCantHandleAction)
+    }
+}
```

Comment:
We already have an extension that starts the activity safely, maybe you should call it and adapt it if needed

Reply:
We might want to do something else, later on, like triggering the share action when it's not possible to open locally. I will add a TODO comment here, or just do it.

Reply:
BTW, opening doesn't work properly yet:
It hilariously tries to open mp4 files in Google Wallet, Samsung Health, or OpenVPN on one of our test devices.

We probably need to pass the proper mime type according to the file extension.